config/config.go

```diff
@@ -302,24 +304,23 @@ func (c Config) String() string {
 	return string(b)
 }
 
-// GetScrapeConfigs returns the scrape configurations.
+// GetScrapeConfigs returns the read-only scrape configurations including
+// the ones from the scrape_config_files.
+// This method:
+// * does not write to c *Config (the pointer receiver is for efficiency).
+// * assumes c.ScrapeConfigs is already validated (in UnmarshalYAML) to avoid writes.
 func (c *Config) GetScrapeConfigs() ([]*ScrapeConfig, error) {
 	scfgs := make([]*ScrapeConfig, len(c.ScrapeConfigs))
 
 	jobNames := map[string]string{}
 	for i, scfg := range c.ScrapeConfigs {
-		// We do these checks for library users that would not call validate in
-		// Unmarshal.
-		if err := scfg.Validate(c.GlobalConfig); err != nil {
```

Comment:
This was causing race (e.g. this writes to `scrapeConfig.TargetLimit = global` and the same `scrapeConfig.TargetLimit` is read on scrape loop)

Reply:
FYI @roidelapluie as [you added this](https://github.com/prometheus/prometheus/commit/599b70a05da832be2746da8a865aef6cec9c246c), in case you rely on it.